elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/Schema.java

```diff
@@ -0,0 +1,290 @@
+        return getMetrics().stream()
+                .filter(metric -> metric.getName().equals(metricName))
+                .findAny()
+                .orElse(null);
+    }
+
+    /**
+     * Returns whether or not an entity field is a metric field.
+     * <p>
+     * A field is a metric field iff that field is annotated by at least one of
+     * <ol>
+     *     <li> {@link MetricAggregation}
+     *     <li> {@link MetricComputation}
+     * </ol>
+     *
+     * @param fieldName  The entity field
+     *
+     * @return {@code true} if the field is a metric field
+     */
+    public boolean isMetricField(String fieldName) {
+        return getEntityDictionary().attributeOrRelationAnnotationExists(
+                getEntityClass(), fieldName, MetricAggregation.class
+        )
+                || getEntityDictionary().attributeOrRelationAnnotationExists(
+                getEntityClass(), fieldName, MetricComputation.class
+        );
+    }
+
+    /**
+     * Returns whether or not an entity field is a simple metric, i.e. a metric field not decorated by
+     * {@link MetricComputation}.
+     *
+     * @param fieldName  The entity field
+     *
+     * @return {@code true} if the field is a simple metric
+     */
+    public boolean isBaseMetric(String fieldName) {
+        if (!isMetricField(fieldName)) {
+            return false;
+        }
+
+        return !getEntityDictionary().attributeOrRelationAnnotationExists(
+                getEntityClass(),
+                fieldName,
+                MetricComputation.class
+        );
+    }
+
+    /**
+     * Constructs a new {@link Metric} instance.
+     *
+     * @param metricField  The entity field of the metric being constructed
+     * @param cls  The entity that contains the metric being constructed
+     * @param entityDictionary  The auxiliary object that offers binding info used to construct this
+     * {@link Metric}
+     *
+     * @return a {@link Metric}
+     */
+    protected Metric constructMetric(String metricField, Class<?> cls, EntityDictionary entityDictionary) {
+        Meta metaData = entityDictionary.getAttributeOrRelationAnnotation(cls, Meta.class, metricField);
+        Class<?> fieldType = entityDictionary.getType(cls, metricField);
+
+        // get all metric aggregations
+        List<Class<? extends Aggregation>> aggregations = Arrays.stream(
+                entityDictionary.getAttributeOrRelationAnnotation(
+                        cls,
+                        MetricAggregation.class,
+                        metricField
+                ).aggregations()
+        )
+                .collect(
+                        Collectors.collectingAndThen(
+                                Collectors.toList(),
+                                Collections::unmodifiableList
+                        )
+                );
+
+        // make sure we have at least 1 aggregation so we can generate static SQL functions
+        if (aggregations.isEmpty()) {
```

Comment:
Based on my previous comment, this check is no longer required.  Instead, it can be moved the constructor of BaseMetric.